docs/StringManifesto.md

```diff
@@ -1124,6 +1124,55 @@ collection of `UInt32`, where values less than `0x80` represent the
 corresponding ASCII character, and other values represent data that is specific
 to the underlying encoding of the string.
 
+### Unsafe String API
+
+While the goal of the String rewrite is to achieve good performance for most usage scenarios without jeopardizing Unicode compliance or type and memory safety, we recognize the importance of being able to squeeze the last bit of performance by dropping down to unsafe APIs.
+
+The following extension makes it possible to access and mutate the content of the String viewed as a contiguous buffer. 
+
+```swift
+extension String {
+    /// The base address of the continuous buffer containing the String elements
+    /// if the String is backed by a contiguous buffer.
+    ///
+    /// There is no guarantee that the String is backed by a contiguous buffer.
+    /// Even if it is, there is no guarantee that the buffer is writable.
+    public var baseAddress: UnsafeRawPointer? { get }
```

Comment:
Without jumping into the whole discussion, this should be an `UnsafeRawBufferPointer?` instead.